server/src/test/java/io/crate/integrationtests/LookupJoinIntegrationTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.integrationtests;
+
+
+
+import static io.crate.testing.Asserts.assertThat;
+
+import org.elasticsearch.test.IntegTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.crate.testing.UseHashJoins;
+import io.crate.testing.UseRandomizedOptimizerRules;
+
+public class LookupJoinIntegrationTest extends IntegTestCase {
+
+    @Before
+    public void setup() throws Exception {
+        execute("create table doc.t1 (id int)");
+        execute("insert into doc.t1 (id) select b from generate_series(1,10000) a(b)");
+        execute("create table doc.t2 (id int)");
+        execute("insert into doc.t2 (id) select b from generate_series(1,100) a(b)");
+        execute("refresh table doc.t1");
+        execute("refresh table doc.t2");
+        execute("analyze");
+        waitNoPendingTasksOnAll();
+    }
+
+    @UseRandomizedOptimizerRules(0)
+    @UseHashJoins(1)
+    @Test
+    public void test_basic_lookup_join() throws Exception {
+        var query = "select count(*) from doc.t1 join doc.t2 on t1.id = t2.id";
+        execute("explain (costs false)" + query);
+        assertThat(response).hasLines(
+            "HashAggregate[count(*)]",
+            "  └ HashJoin[(id = id)]",
```

Comment:
Why do we have still a `HashJoin` here, isn't it just a normal select now with a multiphase ?

Reply:
Since we are retrieving values from both sides, we still need to join. The multiphase-only approach would only work if we retrieved from one side.

Reply:
This is a benchmark resukt 50.000 vs 500.000 from master against this branch:

```
Q: select * from doc.t1 join doc.t2 on t1.id = t2.id
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |      183.414 ±   39.788 |    162.481 |    169.734 |    171.973 |    389.991 |
|   V2    |       39.972 ±   27.423 |     35.147 |     36.344 |     37.167 |    307.615 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               - 128.43%                           - 129.46%
```

Reply:
yep, apologies, I forgot we still need to return columns for both tables, for a moment I thought it simply becomes: `select * from t1 where t1.id IN (SELECT id FROM t2)`.

Reply:
See also https://github.com/crate/crate/issues/15003#issuecomment-1991947230 

> Drop the parent join if only outputs from one side are used

It's another step we can follow up on